docs/porting/custom-target-porting.md

```diff
@@ -0,0 +1,163 @@
+# Porting Mbed OS to a custom board
+
+If you are designing a board similar to an existing Mbed Enabled board, you can use target inheritance to configure your board to meet your needs. In previous versions of Mbed OS (before Mbed OS 5.8), adding a new target required modifying the `mbed-os` directory tree itself, but now you can extend an existing MCU port with a `custom_target.json` file and by adding source files that complete a port outside of `mbed-os`.
+
+For detailed information about all the ways you can configure targets, go to [adding and configuring targets](../reference/adding-and-configuring-targets.html).
+
+## Extending an existing Mbed Enabled MCU target configuration
+
+As an example, consider a situation in which you are creating a new board based on an existing Mbed Enabled board. This tutorial considers a new board called ImaginaryBoard that is based on NRF52840.
+
+ImaginaryBoard is the same as the standard NRF52840 development kit, but you want to remove some unused features, such as AnalogIn, and change the way some pins are connected.
+
+1. Choose a target in Mbed OS that your board can inherit from. 
+
+   If that existing target follows one of the recommended inheritance patterns, such as Family -> MCU -> Board, then it makes it simple to just inherit the MCU target information and add your own board information. In some cases, you may need to inherit a Family or Board and add or replace configurations accordingly.
+
+   In this example, there is an MCU target configuration in `mbed-os/targets/targets.json` that the new board can inherit from. It is called MCU_NRF52840.
+   
+   Below is a short summary of the MCU target. For simplicity, areas have been omitted.
+   
+   From mbed-os/targets/targets.json:
+   
+   ```
+       "MCU_NRF52840": {
+           "inherits": ["Target"],
+           "components_add": […],
+           "core": "Cortex-M4F",
+           "macros": […],
+           "features": […],
+           "device_has": […],
+           "extra_labels": […],
+           "config": {…},
+           "overrides": {…},
+           "OUTPUT_EXT": "hex",
+           "supported_toolchains": ["GCC_ARM", "ARM", "IAR"],
+           "public": false,
+           "detect_code": ["1101"],
+           "bootloader_supported": true
+       }
+   ```
+   
+   For reference, the standard NRF52840 development kit configuration is below. It inherits from the MCU target.
```

Comment:
So in the inheritance you introduced, this is a board, right? It's easier to use the same word here, I think.